keps/sig-node/kubelet-node-endpoint.md

```diff
@@ -0,0 +1,133 @@
+But some node information is currently only available through the `api-server`.
+Having each agent calling the `api-server` to retrieve information may result in an api-server DDOS, which can put in danger the stability of your infrastructure.
+This is the main issue that we want to solve with this KEP.
+
+
+## Proposal
+
+The Kubelet already proposes a set of endpoints (`/pods`, and soon `/metrics/resource/v1*`) that offer useful information for "local" node monitoring. In addition to those endpoints, we would like to introduce a new `/node` endpoint for exposing node resource information. It will allow daemonset-based monitoring solutions to retrieve node information locally (on the same host) like "labels, annotations" instead of using a call to the API server, which will solve some scalability issues in big clusters.
+
+![kubelet-node-endpoint](https://i.ibb.co/khVg8rd/Untitled-Diagram.png)
+
+Kubelet already caches and syncs its corresponding `Node` information from the API-server. The proposed solution opens a new endpoint "/node" that returns the `k8s.io/api/pkg/core/v1.Node` JSON representation of the Kubelet's Node. Only the Kubelet's Node resource is returned.
+
+For consistency the endpoint path "/node" looks similar to the "/pods" endpoint (no API versioning).
+
+## Other Solution
+
+A remediation solution has been already investigated and implemented but requires the deployment of an extra component.
+This component is working as a proxy/cache for queries to the `api-server` done by the daemonset pods.
+This solution resolves the initial problem of reducing pressure on the `api-server`, but in fact, it only moves the problem to a less sensitive component.
+
+This is the solution currently used by the Datadog monitoring solution deployment in Kubernetes via the `cluster-agent`.
+
+### Why /Node endpoint is a better solution
+
+Compared to the previous solution we think that using a `local` communication with the `kubelet` is a better solution and solves that scalability issue that we see with other presented solutions.
+
+With this solution, the load on the `api-server` from the daemonset pods does not depend on the number of nodes. And the load on each kubelet is very low.
+
+We have also seen the same approach, using a daemonset deployment that retrieved kubelet information, in other monitoring solutions.
+For example, GKE is deploying the `prometheus-to-sd` (stackdriver) daemonset pods that retrieve metrics on each kubelet separately.
+
+## Use Cases
+
+Node information, that is currently only available through the `api-server`, provides a lot of context for the metrics generated on application/process running in pods.
+
+### Pod metrics and logs tagging with Node labels or annotations
+
+Monitoring solutions based on the kubelet API ("/pods" and "/metrics") can decorate Pod metrics with additional context information; Information like the "Availability zone", "Provider", "Server type" etc are very useful for investigating issues.
```

Comment:
One thing we might want to consider is exposing some node information through the downward API.  I can imagine other cases where we might, for example, want the address of the node, or the provider id.  This way, we would have more granular control over what is exposed.

It would be helpful to enumerate which portions of the node object you need.  Are the aspects you list all node labels?